aequilibrae/paths/traffic_assignment.py

```diff
@@ -474,12 +480,89 @@ def info(self) -> dict:
                 'Target RGap': self.assignment.rgap_target}
         return info
 
-    def save_skims(self, matrix_name: str, which_ones='final') -> None:
+    def save_skims(self, name: str, which_ones='final', format='omx') -> None:
         """Saves the skims (if any) to the skim folder and registers in the matrix list
 
         Args:
-            matrix_name (:obj:`str`): Name of the file to hold this matrix
+            name (:obj:`str`): Name of the file to hold this matrix
             which_ones (:obj:`str`,optional): {'final': Results of the final iteration, 'blended': Averaged results for
             all iterations, 'all': Saves skims for both the final iteration and the blended ones} Default is 'final'
+            *format* (:obj:`str`, `Optional`): File format ('aem' or 'omx'). Default is 'omx'
         """
-        pass
+
+        mat_format = format.lower()
+        if mat_format not in ['omx', 'aem']:
+            raise ValueError('Matrix needs to be either OMX or native AequilibraE')
+        if mat_format == 'omx' and not has_omx:
+            raise ImportError('OpenMatrix is not available on your system')
+
+        file_name = f'{name}.{mat_format}'
+
+        mats = Matrices()
+        export_name = path.join(mats.fldr, file_name)
+
+        if path.isfile(export_name):
+            raise FileExistsError(f'{file_name} already exists. Choose a different name or matrix format')
+
+        if mats.check_exists(name):
+            raise FileExistsError(f'{name} already exists. Choose a different name')
+
+        avg_skims = self.classes[0].results.skims  # type: AequilibraeMatrix
+
+        # The ones for the last iteration are here
+        last_skims = self.classes[0]._aon_results.skims  # type: AequilibraeMatrix
+
+        names = []
+        if which_ones in ['final', 'all']:
+            if last_skims.names:
+                for core in last_skims.names:
+                    names.append(f'{core}_final')
```

Comment:
```suggestion
        if which_ones in ['final', 'all'] and last_skims.names:
            for core in last_skims.names:
                names.append(f'{core}_final')
```

Reply:
last_skims might be None...

Maybe I should change to an empty matrix?

Reply:
not sure what you mean... last_skims will only be populated if which_ones == final or all?

if so that is what short circuits are for

Reply:
![image](https://user-images.githubusercontent.com/151124/100944706-84194500-354b-11eb-9f40-7fa599c1c0dd.png)
```
>>> def foo():
...     raise ValueError("Dont call me")
...
>>>
>>> a = 7 and foo()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 2, in foo
ValueError: Dont call me
>>> a = False and foo()
>>>
```

Reply:
if the first arg to and is false it never evaluates the second
same with OR except the other way round (if first arg is true, no need to eval the second)

Reply:
Same change as below applies